### PR TITLE
Exclude sessions from CI report

### LINKED ISSUE #3831: allow ILMs to be selected for "count offerings as one"
certainly and edge case, but valid nonetheless - ILMs can offerings too!
therefore, we must allow for them to have their offerings tallied up in the same way as "regular" sessions.

remove the restrictions from the CI sequence block sessions manager currently in place that prohibits ILMs from being flagged as "count as one offering".

### PR DESCRIPTION
**Pull request by stopfstedt**
fixes #3831
refs ilios/ilios#1989

selection_017

### CODE REVIEW

**Comment by jrjohnson** [comment]
Deploy preview for *ilios-frontend* ready!

Built with commit a85a29d614e43dad65724679337f70f8c914a24e

https://deploy-preview-3840--ilios-frontend.netlify.com

File: app/components/curriculum-inventory-sequence-block-session-list.js [hunk 1]
```diff
@@ -24,10 +26,12 @@ export default Component.extend({
 
   loadAttr: task(function * (sequenceBlock, sessions) {
     const linkedSessions = yield sequenceBlock.get('sessions');
-    const linkableSessionsBuffer = yield sessions;
+    const excludedSessions = yield sequenceBlock.get('excludedSessions');
+    const sessionsBuffer = yield sessions;
```
[the remark this answers is not in the document]
Comment (stopfstedt) [reply app/components/curriculum-inventory-sequence-block-session-list.js]: good point. removed.

**Comment by saschaben** [comment]
Looks Good. Please see below for consistency changes to the french translation file. All other stuff looks great.

ilios rapports d inventaire des programmes d etudes 2018-05-29 21-01-26

**Comment by dartajax** [comment]
I found it confusing - hate to say it. When you click "Exclude" at the top, all Sessions get selected. If you then click the "Exclude" check box for an individual session, it appears that the session suddently disappears.

Is that what's happening?
It seems like there's no way to undo it or to know what just happened.

I will try and put it in screen shots below.

image

Now I click on "HIPAA Training". Here is the screen after taking that action. "HIPAA Training" is gone.
How would I get it back or know what just happened?

image

**Comment by stopfstedt** [comment]
@dartajax good find. this should be fixed now.

File: app/components/curriculum-inventory-sequence-block-session-list.js [hunk 2]
```diff
@@ -5,14 +5,15 @@ import { task } from 'ember-concurrency';
...
   sessions: null,
-  linkableSessionsBuffer: null,
+  linkedSessions: null,
+  excludedSessions: null,
+  sessionsBuffer: null,
   classNames: ['curriculum-inventory-sequence-block-session-list'],
   tagName: 'section',
   sortBy: 'title',
 
   init() {
```
Comment (jrjohnson) [inline app/components/curriculum-inventory-sequence-block-session-list.js]: Looks like this isn't ding anything anymore, can `init` just be removed?

File: app/components/curriculum-inventory-sequence-block-session-manager.js [hunk 3]
```diff
@@ -82,7 +116,8 @@ export default Component.extend({
 
   saveChanges: task(function * () {
     let sessions = this.get('linkedSessionsBuffer');
-    yield this.sendAction('save', sessions);
+    let excludedSessions = this.get('excludedSessionsBuffer');
+    yield this.sendAction('save', sessions, excludedSessions);
```
Comment (jrjohnson) [inline app/components/curriculum-inventory-sequence-block-session-manager.js]: Linter will start flagging `sendAction` very soon as an old pattern and possibly a bad idea in the future. Instead the action should be passed in like `{{curriculum-inventory-sequence-block-session-manager save=(action 'save')}}` which makes it a function, then `this.sendAction('save', sessions, excludedSessions);` can be replaced with `this.get('save')(sessions, excludedSessions);`

**Pull request merged**